lib/bridge/IrcBridge.js

```diff
@@ -102,6 +102,21 @@ function IrcBridge(config, registration) {
     );
 
     this._provisioner = null;
+
+    // Cache the mode of each channel, the visibility of each room and the
+    // known mappings between them. When any of these change, any inconsistencies
+    // should be resolved by keeping the matrix side as private as necessary
+    this._visibilityMap = {
+        mappings: {
+            //room_id: ['server #channel1', 'server channel2',...]
+        },
+        channelModes: {
+            // '$server $channel': true | false
```

Comment:
I haven't gone through the bulk of the code yet, I'm just reading the core data structure currently, and this jumps out at me. In order to see if a channel is `+s`, I bet we're going to do:
```js
if (this._visibilityMap.channelModes["irc.freenode.net #chan"]) {
  // stuff
}
```
I don't like that because it isn't clear that this being `true` means the channel is `+s`. Can we rename this to `channelIsSecret` or something instead of the really generic `channelModes` (of which it doesn't actually represent the set of channel modes, just a very specific one, `+s`). This would mean the equiv is:
```js
if (this._visibilityMap.channelIsSecret["irc.freenode.net #chan"]) {
  // stuff
}
```
which is then immediately clear what is going on.